pkg/minishift/instancedata/instancedata.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package instancedata
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+)
+
+// Variable to be used anywhere
+var Data = new(InstanceData)
+
+type InstanceData struct {
+	FilePath string `json:"-"`
+	OcPath   string
+}
+
+func (m *InstanceData) Initialize(path string) {
```

Comment:
the better approach is to have a a factory method _newInstanceConfig(string path)_ which creates and configures the instance. Having this initialize method in this form causes two problems:

* It is not clear that you need to call _Initialize_ at all on _Data_. This can lead to usage of the instance even though it is not initialized.
* Secondly, the config path should be internal and unchangable. What if initialize gets called more than once, potentially changing the path.

See also http://www.golangpatterns.info/object-oriented/constructors. You want this object to be safe to use and avoid accidental misuse as much as possible